mlir/lib/Dialect/SCF/Transforms/TileUsingInterface.cpp

```diff
@@ -1072,14 +1137,21 @@ mlir::scf::tileConsumerAndFuseProducersUsingSCF(
       continue;
 
     if (yieldReplacement) {
+      // Reconstruct and yield all opResult of fusableProducerOp by default. The
+      // caller can specific which one to yield by designating optional argument
+      // named `yieldResultNumber` of `yieldReplacementForFusedProducer`.
+      Operation *fusableProducerOp = fusableProducer.getOwner();
       if (failed(yieldReplacementForFusedProducer(
               rewriter, candidateSliceOp, fusedResult.value(), loops))) {
         return rewriter.notifyMatchFailure(
-            fusableProducer.getOwner(), "failed to replacement value for this "
-                                        "oepration from within the tiled loop");
+            fusableProducerOp, "failed to replacement value for this "
+                               "operation from within the tiled loop");
+      }
+      for (const auto &result : fusableProducerOp->getResults()) {
```

Comment:
Nit: this could probably be easier as 
```
for (auto [index, result]: llvm::enumerate(fusableProducerOp->getResults())) {
  origValueResultNumber[result] = loops.front()->getNumResults() - fusableProducerOp->getNumResults() + index;
}

Reply:
Sounds good. changed :)